### ISSUE TITLE
Loss of touch control for some time (few suspend on/off cycles returns control so far)

### ISSUE THREAD

**Issue by AlanSP1** [koreader/koreader#5333]
* KOReader version: koreader-kobo-arm-kobo-linux-gnueabihf-v2019.08-25-g3a957d7_2019-08-20

* Device: Kobo Aura One

OK, I'll try to explain what happened and what I think is important, but, some things may be relevant, or may not be relevant, but I'll try to include everything I remember.

In both times reader was started with opening cover. So, it may be a factor (I think it is, but not sure).

First time, after I opened cover, reader waked but for some strange reason opened image that was on page. Not sure why, I sure didn't tried to open image, maybe related to some problems with cover and touch screen (KA1 has flush bezel, so there might be some connection, or something that happens before waking. And, just in case, I suspend device with button, not with cover).

It just looked like this: I open cover, device wakes up, "selects" image (I think I saw short "flash" around image, not sure), image is opened.

After that I was looking at image. Well, just to remind, I had crash problem described here: https://github.com/koreader/koreader/issues/5154 This was a very similar, except now light was off.

Touch wasn't working, I couldn't close image and return to document. So I started using power button to suspend on/off device and, after 5-6 tries (not sure how many), touch control returned.

After that, everything worked out as normal.

Second time, device again waked with opening cover, but after I opened it, low battery warning message appeared and again ... loss of touch control.

Again, after some suspending/waking cycles, touch started working. But, at first wonky (I couldn't actually dismiss low battery warning at first, only after some time). After that, again, everything worked fine.

So, I guess, maybe it is cover waking and immediate widget activation that causes loss of touch control for some time, or for some suspend/wake up cycles. After that, everything works again.

**Comment by NiLuJe** [comment]
*Possibly* similar with #4431 ?

**Comment by AlanSP1** [comment]
Not sure if it is same issue. In my case, there was no dictionary, but it is widget, so it is similar.

Also, not sure if my taps were interpreted as long taps. And, I don't think I moved finger while suspending/waking.

But I tried to access menu (swipe down), didn't worked for me. Only power button worked, or at least, that's what I found out.

For the moment, I think it is something with widgets and possibly cover. But yes, touchscreen on KA1 is not something to write home about.

**Comment by NiLuJe** [comment]
IIRC, showing an image requires a long-tap, which is why i thought is this ;).

**Comment by poire-z** [comment]
> So I started using power button to suspend on/off device and, after 5-6 tries (not sure how many), touch control returned

When I get that kind of issue on my GloHD:
1. power button to resume > no touch
2. power button to suspend
3. power button to resume

I need to wait 15 seconds between 2. and 3. for the Kobo to really get to sleep, so that the next power button really starts from a clean suspend state and get a chance to reset the touch layer.
I'd like to know it that works for you too (to avoid the 5-6 tries at least). I always assumed it was the IR grid layer, but if you get that too with a capacitive screen, it might not be a hardware issue.

It happens sometimes that one of the power button press is kind of not processed and stays in a queue, and is only processed on the next power button press - so then it resumes and immediately goes back to sleep (screensaver cover disappear and immediately reappear...)
I still suspect it's something in our event processing stack, like the thing staying in some "hold" state, because sometimes, when I resume, my next touch extend some selection...

**Comment by NiLuJe** [comment]
@poire-z: That's probably the `sleep` involved in the suspend code eating up events, there's either an issue or a PR with an originally SleepCover-centered  patch that implements it differently to avoid that.

**Comment by AlanSP1** [comment]
Yes, you're right here but... from my point of view, I didn't touched screen (but cover might did some "magic" of its own, I'm not sure).

From my perspective, I just opened cover (without any screen touching), device "zoomed" into image and voila, I was stuck for some time without touch control.

So, it may be that cover caused this phantom short tap that was interpreted as long, thus opening image, but I can't be sure..

**Comment by AlanSP1** [comment]
> I need to wait 15 seconds between 2. and 3. for the Kobo to really get to sleep, so that the next power button really starts from a clean suspend state and get a chance to reset the touch layer.
I'd like to know it that works for you too (to avoid the 5-6 tries at least). I always assumed it was the IR grid layer, but if you get that too with a capacitive screen, it might not be a hardware issue.

If that happens again, I'll sure test for that. Who knows, may be that.

Just to mention, didn't had something like that when I turn device with button (sometimes I just don't cover it, i.e. when I'll read pretty soon, don't need to transport reader etc, I don't close cover). So, cover may be a factor, or may just coincide with widget firing immediately after waking device on.

**Comment by AlanSP1** [comment]
> I still suspect it's something in our event processing stack, like the thing staying in some "hold" state, because sometimes, when I resume, my next touch extend some selection...

It can be something like that, I had problems second time, touch started working, but wonky, as I described. Only after few touches (again, not sure, but like 5-6 times) it started working as it should.

**Comment by retrue** [comment]
Hi, in my Kobo Mini something similar is happening since a couple of days ago.
Sometimes the screen doesn't respond to touch. I touch the screen to turn page and it loads again the same page, after two or three tries I am able to turn page but then I have the same problem to turn page again, the other touch controls don't work, like opening menus, etc.
If I push the power button to sleep and then back, the device works again fine for a while.

**Comment by AlanSP1** [comment]
It seems to me like completely different issue, but of course, can't be sure.

In my case, loss of touch is "one time event", after I recover from it, there's no problems with touch. Of course, till next trigger, but in my two cases it always was waking device up (and with cover, but it may be coincidence).

**Comment by cramoisi** [comment]
I got this issue since day 1 with my KA1 and KOReader after wake it up from sleep (with cover). The screen doesn't respond to touch and I need to wait or to make a big gesture (like swipe the screen with all my finger) to take the control back. 

It happens exactly like @AlanSP1 said is the post just above.

**Comment by aatdevries** [comment]
I have the same issue with my Kobo Aura One. This last comment helped a lot, swiping will wake up the device most of the time, but sometimes it will change font size or create a screen shot. I am willing to troubleshoot, if someone can give me instructions how to do this.

**Comment by NiLuJe** [comment]
That'd imply running with *verbose* debug logging enabled (FM, Tools > More Tools > Developer Tools).

This'll dump *a lot* of stuff to the crash.log, including details about input events.

**Comment by aatdevries** [comment]
Sorry, I'm new here, so this might be a stupid question. In the Tools > More Tools menu I don't see Developer Tools (I do see battery statistics, synchronise time, keep alive ...)

**Comment by Frenzie** [comment]
Only in the file browser atm

**Comment by aatdevries** [comment]
See attached my crash.log. In the last couple of days I didn't have time to do any reading, but I've opened the sleepcover a couple of times and tried different gestures to make the screen responsive again. Hopefully the log will provide useful information for you guys. If there is anything you would like me to try on the device, please let me know.

[crash.log](https://github.com/koreader/koreader/files/5710217/crash.log)

**Comment by NiLuJe** [comment]
Meh, broken timestamps that don't tick during suspend, which probably explains *some* of it.

I'm not entirely familiar with the intricacies of the input system, but we could/should probably reset the state tracking on suspend/resume.
